scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -510,14 +511,52 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
   def groupBy[K](
     f: T => K
   )(implicit kcoder: Coder[K], vcoder: Coder[T]): SCollection[(K, Iterable[T])] =
+    groupMap(f)(identity)
+
+  /**
+   * Return an SCollection of grouped items. Each group consists of a key and a sequence of
+   * elements transformed into a value of type `U`. The ordering of elements within each group is not guaranteed,
+   * and may even differ each time the resulting SCollection is evaluated.
+   *
+   * It is equivalent to groupBy(key).mapValues(_.map(f)), but more efficient.
+   *
+   * @group transform
+   */
+  def groupMap[K, U](f: T => K)(
```

Comment:
How does this compare to the more idiomatic `.map(x => (f(x), g(x))).groupByKey`?

Reply:
Same thing applies for this example one step less of conversion/serialization. However the idea behind this method is to try to mitigate other issues like:

```scala
groupBy(key).mapValues(_.map(f))
```

Users will incur in an extra copy of the list which depending on the size might be significant perf penalty.